rest-api/dao/base_dao.py

```diff
@@ -331,8 +330,6 @@ def _insert_with_random_id(self, obj, fields):
       except IntegrityError, e:
         # SQLite and MySQL variants of the error message, respectively.
         if 'UNIQUE constraint failed' in e.message or 'Duplicate entry' in e.message:
-          if 'external_id' in e.message:
```

Comment:
So I would retain this logic -- this conflict can happen if two requests with the same external ID happen around the same time. But instead of raising a Conflict, I'd have the client pass in a function that can be called to determine what to do / return... and have participant_dao pass in something that does the lookup by external ID.

Reply:
done. (If I understood you correctly 😄 ).